lib/ssl/src/ssl_gen_statem.erl

```diff
@@ -302,6 +311,32 @@ socket_control(dtls_gen_connection = Connection, {PeerAddrPort, Socket}, [Pid|_]
 	    {error, Reason}
     end.
 
+prepare_connection(
+    #state{
+        static_env = #static_env{
+            transport_cb = gen_tcp,
+            socket = Socket
+        },
+        ssl_options = #{use_ktls := true},
+        socket_options = #socket_options{
+            mode = Mode,
+            packet = Packet,
+            packet_size = PacketSize,
+            header = Header,
+            active = Active
+        },
+        handshake_env = #handshake_env{renegotiation = {false, first}},
+        connection_env = #connection_env{user_application = {_Mon, Pid}},
+        connection_states = ConnectionStates,
+        start_or_recv_from = StartFrom
+    } = State,
+    _Connection
+) ->
+    gen_tcp:controlling_process(Socket, Pid),
+    set_ktls(tls_1_3_aes_gcm_256, ConnectionStates, Socket),
```

Comment:
Would it not be more logical to first set the socket options and then hand over the control?